src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -751,22 +783,32 @@ namespace AppInstaller::CLI::Workflow
                     // Output using the local PackageName instead of the name in the manifest, to prevent confusion for packages that add multiple
                     // Add/Remove Programs entries.
                     // TODO: De-duplicate this list, and only show (by default) one entry per matched package.
-                    table.OutputLine({
+                    InstalledPackagesTableLine line(
                          installedVersion->GetProperty(PackageVersionProperty::Name),
                          match.Package->GetProperty(PackageProperty::Id),
                          installedVersion->GetProperty(PackageVersionProperty::Version),
                          availableVersion,
-                         shouldShowSource ? sourceName : ""s
-                    });
-                   
-                   
+                         shouldShowSource ? sourceName : Utility::LocIndString()
+                    );
+
+                    auto pinnedState = ConvertToPackagePinnedStateEnum(installedVersion->GetMetadata()[PackageVersionMetadata::PinnedState]);
+                    bool requiresExplicitUpgrade = m_onlyShowUpgrades &&
+                        (pinnedState == PackagePinnedState::PinnedByManifest);

```

Comment:
I think you are going to want to either create a semantic check function here, or always use `pinnedState != Unpinned` to enable other pinning reasons to automatically be included.